### PR TITLE
Adding crd api doc generation

### LINKED ISSUE #7: Autogenerate CRD docs from code

### PR DESCRIPTION
**Pull request by souleb**
fixes #7

### CODE REVIEW

File: hack/api-docs/config.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      "typeMatchPrefix": "^k8s\\.io/api/core/v1\\.NodeAffinity$",
+      "docsURLTemplate": "https://godoc.org/k8s.io/api/core/v1#NodeAffinity"
+    },
+    {
+      "typeMatchPrefix": "^k8s\\.io/api/core/v1\\.Toleration$",
+      "docsURLTemplate": "https://godoc.org/k8s.io/api/core/v1#Toleration"
+    },
+    {
+      "typeMatchPrefix": "^k8s\\.io/(api|apimachinery/pkg/apis)/",
+      "docsURLTemplate": "https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#{{lower .TypeIdentifier}}-{{arrIndex .PackageSegments -1}}-{{arrIndex .PackageSegments -2}}"
```
Comment (rollandf) [inline hack/api-docs/config.json:28]: This link seems dead

File: Makefile [hunk 2]
```diff
@@ -79,6 +89,16 @@ branch-helm-docs: download-branch-helm-chart helm-docs
 .PHONY: local-helm-docs
 local-helm-docs: copy-local-helm-chart helm-docs
 
+.PHONY: download-branch-api
+download-branch-api: | $(CRD_API_DEP_ROOT)
+	curl -sL ${BRANCH_REPO_URL} \
+	| tar -xz -C ${CRD_API_DEP_ROOT}
+
+.PHONY: gen-crd-api-docs
+gen-crd-api-docs: | $(GEN_CRD_API_REFERENCE_DOCS)
```
Comment (rollandf) [inline Makefile:98]: Should we have a dependency on download-branc-api target?

**Comment by souleb** [comment]
see https://github.com/Mellanox/network-operator-docs/actions/runs/10354494117/artifacts/1802382526. Everything looks good.

**Comment by rollandf** [comment]
Looks great!
We need to find a solution for the inner links that are not working

**Comment by souleb** [comment]
Thanks @rollandf, all the links work now!

**Review by rollandf — APPROVED** [review]
Thanks!

**Pull request merged**